### PR TITLE
Fix #113: Refactor getSitePackagesPath to use python discovery

### LINKED ISSUE #113: Strange assumptions made about poetry virtualenvs
### Describe the bug

Poetry build is making very strange assumptions about the nature of the virtualenv which poetry creates.

### Current behavior

As far as I'm aware I have a completely normal poetry setup. I'm using pyenv to manage my virtual environments. When I run `bi poetry install` i get the following output:
```
[Error] failed extracting python site package form the following virtual env "/home/rvodden/.pyenv/versions/3.10.7/envs/library"
```

### Reproduction steps

install pyenv
create a virtualenv using pyenv
create a poetry project in that virtualenv
run `bi poetry install`

### Expected behavior

I'd expect `bi` to use correct semantics to establish the location of site-packages. For example, this command:

```
poetry run python -c 'import site; print(site.getsitepackages())'
```

will reliably report the location of `site-packages`.  I don't really follow the logic which is employed to extract the site-packages location here:
https://github.com/jfrog/build-info-go/blob/main/utils/pythonutils/poetryutils.go#L211-L217

It looks to be making very specific assumptions about how virtualenvs are arranged which at first glance would appear to be more likely to be incorrect than correct in the majority of circumstances

### Operating system type and version

ubuntu 20.04 i686

### PR DESCRIPTION
**Pull request by rvodden**
- [X] All [tests](https://github.com/jfrog/build-info-go#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [X] All [static analysis checks](https://github.com/jfrog/build-info-go/actions/workflows/analysis.yml) passed.
- [X] This pull request is on the dev branch.
- [X] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

**Comment by rvodden** [comment]
I have read the CLA Document and I hereby sign the CLA

File: utils/pythonutils/poetryutils.go [hunk 1]
```diff
@@ -164,66 +166,28 @@ func extractPoetryDependenciesFiles(srcPath string, cmdArgs []string, log utils.
...
-				return "", nil
-			}
-			// If found, return the virtual env path
-			return pattern.MatchedResults[1], nil
-		},
-	}
-	virtualEnvPath, errorOut, _, err := gofrogcmd.RunCmdWithOutputParser(installCmd, true, &virtualEnvNameParser)
+func getSitePackagesPath() (sitePackagesPath string, err error) {
+
+	siteCmd := utils.NewCommand("poetry", "run", []string{"python", "-c", "import sysconfig;  print(sysconfig.get_paths()['purelib']])"})
```
Comment (talarian1) [inline utils/pythonutils/poetryutils.go:184]:
```suggestion
	siteCmd := utils.NewCommand("poetry", "run", []string{"python", "-c", "import sysconfig;  print(sysconfig.get_paths()['purelib'])"})
```

**Review by talarian1 — CHANGES_REQUESTED** [review]

**Comment by rvodden** [comment]
> Hi @rvodden, thanks a lot for your contribution.
> I suggest we will wait with those changes, as for Poetry 1.2 and above the file 'direct_url.json' is not created anymore.
> You can see this [discussion](https://github.com/orgs/python-poetry/discussions/6869) for more details.
> The only usage for the getSitePackagesPath logic is getting those files.
> Sorry for the inconvenience and thanks again!

aha great! I must admit i was thinking of authoring a poetry plugin which would output buildinfo... would that be interesting as an alternative to having to install jf-cli ?

**Comment by talarian1** [comment]
Thanks again for your interest in this, and for your effort.
I'm closing this for now, as we discussed [here](https://github.com/orgs/python-poetry/discussions/6869) we will need a new implementation to solve this issue.

**Pull request closed without merging**